lib/Sema/DerivedConformanceParameterized.cpp

```diff
@@ -400,9 +431,34 @@ static Type deriveParameterized_Parameters(DerivedConformance &derived) {
   return derived.getConformanceContext()->mapTypeIntoContext(parametersType);
 }
 
+static Type deriveParameterized_VectorSpace(Identifier id,
+                                            DerivedConformance &derived) {
+  auto &TC = derived.TC;
+  auto parentDC = derived.getConformanceContext();
+  auto nominal = derived.Nominal;
+  auto &C = nominal->getASTContext();
+
+  auto parametersType = getParametersStructDecl(nominal).first
+      ->getDeclaredInterfaceType();
+  auto vectorSpaceType = DependentMemberType::get(parametersType, id);
+//  vectorSpaceAlias->setUnderlyingType(vectorSpaceType);
+//  vectorSpaceAlias->copyFormalAccessFrom(nominal);
+//  vectorSpaceAlias->setImplicit();
+//  C.addSynthesizedDecl(vectorSpaceAlias);
+//  auto typealiasType = vectorSpaceAlias->getDeclaredInterfaceType();
+  llvm::outs() << "!!!!!!!! DERIVED " << vectorSpaceType << '\n';
+  llvm::outs().flush();
+  return derived.getConformanceContext()->mapTypeIntoContext(vectorSpaceType);
+}
+
 Type DerivedConformance::deriveParameterized(AssociatedTypeDecl *requirement) {
   if (requirement->getBaseName() == TC.Context.Id_Parameters)
     return deriveParameterized_Parameters(*this);
+  if (requirement->getBaseName() == TC.Context.Id_TangentVector)
+    return deriveParameterized_VectorSpace(TC.Context.Id_TangentVector, *this);
+  if (requirement->getBaseName() == TC.Context.Id_CotangentVector)
+    return deriveParameterized_VectorSpace(TC.Context.Id_CotangentVector,
+                                           *this);
```

Comment:
Don't worry about synthesizing requirements defined on `Differentiable`: those will fall out of `Differentiable` derived conformances. The newly added lines here won't trigger.